src/wazuh_modules/vulnerability_scanner/qa/test_efficacy_log.py

```diff
@@ -157,98 +157,94 @@ def run_process_and_monitor_log(request, run_on_end):
-        LOGGER.info(f"Process initialized")
-
-    expected_json_files = sorted(Path(test_folder).glob("expected_*.out"))
-    expected_lines = []
-    # Read expected output if it exists, this is an json with and array of lines.
-    for expected_json_file in expected_json_files:
-        # Parse json and add the string elements of te array to the expected lines
-        json_data = json.load(open(expected_json_file))
-        for line in json_data:
-            expected_lines.append(line)
-
-    LOGGER.debug(f"Expected lines: {expected_lines}")
-    quantity_expected_lines = len(expected_lines)
-    LOGGER.debug(f"Quantity expected lines: {quantity_expected_lines}")
-
-    found_lines = {line: False for line in expected_lines}
-    timeout = 10
-
-    json_files = sorted(Path(test_folder).glob("input_*.json"))
-
-    for json_file in json_files:
-        LOGGER.debug(f"Running test {json_file}")
-        with open(json_file) as f:
-            # Set the output file
-            file = str(json_file)
-
-            # Parse json file and print the data
-            json_data = json.load(open(file))
-
-            # Set the output folder
-            output_folder = str(test_folder)
-
-            # Set the output file
-            output = str(json_file).replace(".json", ".bin")
-
-            # Convert the json data to flatbuffer
-            json2binary(file, output_folder)
-
-            # Read the flatbuffer data
-            with open(output, "rb") as f:
-                flatbuffer_data = f.read()
-
-            LOGGER.debug("Sending flatbuffer data")
-
-            # After start to read lines, send the flatbuffer data
-            sendflatbuffer_to_unixsocket(flatbuffer_data)
-
-    # Wait until the scan is finished, the scan is finished with this line: "Vulnerability scan for package '*' on Agent '*' has completed." using regex.
-    if not test_folder.name == '000':
-        regex = r"Event type: (.*) processed"
-        found = find_regex_in_file(regex, log_file, len(expected_json_files))
-        regex = r"Discarded event: DB query not synced"
-        found = found or find_regex_in_file(regex, log_file, len(expected_json_files))
-        assert found, "The scan is not finished, some events were not processed"
-        LOGGER.info(f"Scan finished, all events were processed")
-
-    basetimeout = timeout
-    for expected_line in expected_lines:
-        while not found_lines[expected_line]:
-            LOGGER.debug(f"Waiting for log line: {expected_line}")
-            if timeout < 8*basetimeout:
-                tail_log(log_file, expected_lines, found_lines, timeout)
-                timeout = 1.5*timeout
-            else:
-                LOGGER.error(f"Timeout waiting for log line: {expected_line}")
-                timeout = basetimeout
-                break
-
-    process.terminate()
+    with subprocess.Popen(command) as process:
```

Comment:
A little bit out of scope, but the changes below are just the indentation added

This improves the management of the process and prevents leaks for not calling terminate

Reply:
Indeed out of scope, it was useful while running the tests, but this is present in 4.8.0